File: LeetCode/Python/src/Reverse_Integer.py
"""
@author - Anirudh Sharma
"""
import sys


def reverse(x: int) -> int:
    isNegative = False
    if x < 0:
        isNegative = True
        x = -x
    reversedNumber = 0
    while x:
        reversedNumber = reversedNumber * 10 + x % 10
        x //= 10
    if reversedNumber >= 2 ** 31 - 1 or reversedNumber <= -2 ** 31:
        return 0
    return -reversedNumber if isNegative else reversedNumber


if __name__ == '__main__':
    print(reverse(2342))
    print(reverse(-4345))
    print(reverse(340))
    print(sys.maxsize)

File: LeetCode/JavaScript/src/reverse_integer.js
/**
 * @author Anirudh Sharma 
 */
var reverse = function (x) {
    let isNegative = false;
    if (x < 0) {
        isNegative = true;
        x = -x;
    }
    let reverse = 0;
    while (x > 0) {
        reverse = reverse * 10 + x % 10;
        x = parseInt(x / 10);
    }
    if (reverse >= Math.pow(2, 31) - 1 || reverse <= Math.pow(-2, 31)) {
        return 0;
    }
    return isNegative ? -reverse : reverse;
};

console.log(reverse(342));
console.log(reverse(-430));
console.log(reverse(242353400));